compiler/src/dotty/tools/repl/ReplDriver.scala

```diff
@@ -243,7 +243,7 @@ class ReplDriver(settings: Array[String],
       unfusedPhases(using ctx).collectFirst { case phase: CollectTopLevelImports => phase.imports }.get
 
     implicit val state = {
-      val state0 = newRun(istate)
+      val state0 = newRun(istate, parsed.reporter)
       state0.copy(context = state0.context.withSource(parsed.source))
```

Comment:
Btw, before I wired the reporter through newRun I was adding it here with `.fresh.setReporter(reporter)` but that was breaking something to do with class-loading, which seemed complicated.  I'm not sure if that's expected and if anyone knows why that happened.